### PR TITLE
make packages.repositories.url.repository be the repository url instead of a .repo file

### PR DESCRIPTION
**Pull request by rcernich**
Implements #262.

Signed-off-by: rcernich <rcernich@redhat.com>

### CODE REVIEW

File: cekit/descriptor/packages.py [hunk 1]
```diff
@@ -80,8 +86,8 @@ def _create_repo_object(self, repository):
                        "for more details." % repository)
         descriptor = {}
         descriptor['name'] = repository
-        descriptor['url'] = {}
-        descriptor['url']['repository'] = self._get_repo_url(descriptor)
+        descriptor['url'] = {'repository': 'dummy'}
```
Comment (dbecvarik) [inline cekit/descriptor/packages.py]: why dummy here?

File: cekit/descriptor/packages.py [hunk 2]
```diff
@@ -111,5 +117,18 @@ def _get_repo_url(self, descriptor):
...
+                .copy(os.path.join(target_dir, self._descriptor['filename']))
+        else:
+            config = configparser.ConfigParser()
+            sectionName = self._descriptor['name']
+            config.add_section(sectionName);
+            config.set(sectionName, 'name', sectionName);
+            config.set(sectionName, 'baseurl', self._descriptor['url']['repository'])
+            config.set(sectionName, 'enabled', 1)
+            if 'gpg' in self._descriptor['url']:
+                config.set(sectionName, 'gpgkey', self._descriptor['url']['gpg'])
```
Comment (dbecvarik) [inline cekit/descriptor/packages.py]: why have gpg url and do not embed it in descriptor too?

File: cekit/descriptor/packages.py (hunk 1 above)
Comment (rcernich) [reply cekit/descriptor/packages.py]: Just a place holder so the fetch logic is triggered.

File: cekit/descriptor/packages.py (hunk 2 above)
Comment (rcernich) [reply cekit/descriptor/packages.py]: It does get embedded in the descriptor.

The following:
```yaml
packages:
  repositories:
  - name: rhos
    url: 
      repository: "http://download.devel.redhat.com/released/RHEL-7/7.5/Server/x86_64/os/"
      gpg: "http://download.devel.redhat.com/released/RHEL-7/7.5/Server/x86_64/os/RPM-GPG-KEY-redhat-release"
```

produces:
```ini
[rhos]
name = rhos
baseurl = http://download.devel.redhat.com/released/RHEL-7/7.5/Server/x86_64/os/
enabled = 1
gpgkey = http://download.devel.redhat.com/released/RHEL-7/7.5/Server/x86_64/os/RPM-GPG-KEY-redhat-release
gpgcheck = 1
```

**Comment by dbecvarik** [comment]
I'll close this as is probably stale now.

**Pull request closed without merging**